src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerInitializationHostService.cs

```diff
@@ -0,0 +1,118 @@
+            {
+                await InitializeAssignmentContext();
+            }
+        }
+
+        private async Task InitializeAssignmentContext()
+        {
+            var startContext = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerStartContext);
+
+            // Container start context is not available directly
+            if (string.IsNullOrEmpty(startContext))
+            {
+                _logger.LogInformation("AssignmentContext not available in ContainerStartContext");
+                // Check if the context is available in blob
+                var sasUri = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerStartContextSasUri);
+
+                if (!string.IsNullOrEmpty(sasUri))
+                {
+                    _logger.LogInformation("AssignmentContext ContainerStartContextSasUri available");
+                    startContext = await GetAssignmentContextFromSasUri(sasUri);
+                }
+            }
+
+            if (!string.IsNullOrEmpty(startContext))
+            {
+                _logger.LogInformation("Assigning HostAssignmentContext.");
+
+                var encryptedAssignmentContext = JsonConvert.DeserializeObject<EncryptedHostAssignmentContext>(startContext);
+                var containerKey = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerEncryptionKey);
+                var assignmentContext = encryptedAssignmentContext.Decrypt(containerKey);
+                if (_instanceManager.StartAssignment(assignmentContext))
+                {
+                    _logger.LogInformation("Assign HostAssignmentContext success");
+                }
+                else
+                {
+                    _logger.LogError("Assign HostAssignmentContext failed");
+                }
+            }
+            else
+            {
+                _logger.LogInformation("Waiting for /assign to receive AssignmentContext");
+            }
+        }
+
+        private async Task<string> GetAssignmentContextFromSasUri(string sasUri)
+        {
+            try
+            {
+                return await Read(sasUri);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, $"Error calling {nameof(GetAssignmentContextFromSasUri)}");
+            }
+
+            return string.Empty;
+        }
+
+        // virtual for unit testing
+        public virtual async Task<string> Read(string uri)
+        {
+            var cloudBlockBlob = new CloudBlockBlob(new Uri(uri));
+            if (await cloudBlockBlob.ExistsAsync(null, null, _cancellationToken))
+            {
+                return await cloudBlockBlob.DownloadTextAsync(null, null, null, null, _cancellationToken);
+            }
+
+            return string.Empty;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Stopping LinuxContainerInitializationService.");
```

Comment:
I don't think we need this log statement :)

Reply:
removed :)